hazelcast-jet-core/src/main/java/com/hazelcast/jet/Processors.java

```diff
@@ -375,11 +377,13 @@ public static ProcessorSupplier streamFiles(@Nonnull String watchedDirectory) {
      *
      * @param watchedDirectory The directory where we watch files
      * @param charset charset to use to decode the files, or {@code null} to use UTF-8
+     * @param glob the file filter, see {@link
+     *             java.nio.file.FileSystem#getPathMatcher(String) getPathMatcher()}
      */
     public static ProcessorSupplier streamFiles(
-            @Nonnull String watchedDirectory, @Nullable Charset charset
+            @Nonnull String watchedDirectory, @Nullable Charset charset, @Nullable String glob
     ) {
-        return StreamFilesP.supplier(watchedDirectory, charset == null ? null : charset.toString());
+        return StreamFilesP.supplier(watchedDirectory, charset == null ? null : charset.toString(), glob);
```

Comment:
Here we have another place where we accept `null` and substitute defaults. Maybe not in this PR, but we should change this.